### ISSUE TITLE
`entrypoint` attemps a cross-mount `rename(2)`

### ISSUE THREAD

**Issue by bbguimaraes** [kubernetes/test-infra#18640]
Our [canary job](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/10655/rehearse-10655-pull-ci-openshift-release-master-config/1290580433699868672) started failing after updating the `test-infra` images to `v20200804-7b7bfd1dd9`.

The specific error is:

```
{
  "component": "entrypoint",
  "error": "could not move marker file to destination path (/logs/marker-file.txt): rename /tmp/temp-marker667309304 /logs/marker-file.txt: invalid cross-device link",
  "file": "prow/entrypoint/run.go:82",
  "func": "k8s.io/test-infra/prow/entrypoint.Options.Run",
  "level": "error",
  "msg": "Error writing exit code to marker file",
  "severity": "error",
  "time": "2020-08-04T09:06:01Z"
}
```

This most likely is due to https://github.com/kubernetes/test-infra/pull/18278, specifically:

> Entrypoint's Mark() function creates a temp file outside the directory of the desired marker file. This is necessary to prevent concurrency bugs with fsnotify's addWatch().

`/logs` in the container is:

```yaml
volumes:
# …
- emptyDir: {}     
  name: logs             
```

And the resulting file systems:

```
sh-4.2$ mount | awk '$3 ~ /^\/(|tmp|logs)$/'
overlay on / type overlay (rw,relatime,context="system_u:object_r:container_file_t:s0:c4,c9",…)
/dev/sda1 on /logs type xfs (rw,relatime,seclabel,attr2,inode64,grpquota)
```

From a cursory look, we need to either:
- revert to using the same directory, possibly a sub-directory to avoid the watch issues
- use a less restrictive function/syscall to move the marker 

/cc @anthonyrtong

**Comment by BenTheElder** [comment]
temp file + rename is the only way to do atomic write so the second option is likely invalid.

rename has to be on the same filesystem.

**Comment by anthonyrtong** [comment]
Yes, this is the issue.  Changing this to use the sub-directory would probably be the best option and would hopefully still be able to avoid the concurrency issues.

**Comment by alvaroaleman** [comment]
This seems to be fixed by #18642. Thanks @anthonyrtong !
/close
